### ISSUE TITLE
'OSError : Out of pty devices' on Centos 7.5 for non-root users

### ISSUE THREAD

**Issue by maliky** [pypa/pipenv#2224]
When running as non root user on centos 7.5, `pipenv shell` returns an `OS Error: Out of pty devices`

There are 4096 pty available on the machine and only 22 are used.

A `strace` shows a permission error on `/dev/ptmx` but the file exist and has +rw rights for the group of the user making the call.  And every thing works well when `pipenv shell` is run as root.

`$ python -m pipenv.help`.
https://gist.github.com/maliky/01249e028943a350512d70d697038017

------------

##### Expected result
The shell in the virtual env for a non root user.

##### Actual result

the console's trace after `pipenv shell` 
https://gist.github.com/maliky/f17c7e73d96cccbb90d9efac15c0957f
 
the excerpt of strace showing the permission error
https://gist.github.com/maliky/2a15911790e5642688f0345eb8e28506
```
dev/ptmx", O_RDWR)               = -1 EACCES (Permission denied)
open("/dev/ptyp0", O_RDWR)              = -1 ENOENT (No such file or directory)
open("/dev/ptmx", O_RDWR) = -1 EACCES (Permission denied)
```

The full `strace pipenv shell`.  
https://gist.github.com/maliky/ae17fc015913907e37ce058ae27d569a

##### Steps to replicate
I'm running centos 7.5

- create a normal user
- then log in his account
 \#su -l user

-  install pipenv for the user
$ pip3 install pipenv

- create a virutal env with pipenv
$ pipenv --three

- try to open the shell in the virtual env
$ pipenv shell

**Comment by techalchemy** [comment]
Based on [this blog post I found when googling this error](http://adam8157.info/blog/2017/08/fix-out-of-pty-devices/) you need to add a user to the `tty` group if you want to jump from a root shell to that users shell and then fork a subshell. I think this is probably a RHEL/CentOS/SELinux feature but I haven’t tested it anywhere so it could just be universally true. I wouldn’t generally recommend logging in as root in the first place, you won’t encounter this if you login as the target user to start with. `su` is not perfect and it is usually recommended to do things with sudo

**Comment by maliky** [comment]
> found when googling this error 

Lucky you, and me.  :) 
Your google search history, language and localization, probably made it easier for you to find the pertinent link.   I googled this before, read some stackexchange threads, let time pass (a few months), did some search in my os manual, asked my host.  I even remember thinking of doing what this blog recommend without doing it because I wanted to read about it first.  Thanks to you I did it today and my issue is cleared.

As for logging as root in the first place you are right.   Merci !
